### DISCUSSION TITLE
Postprocessors do not save to csv file with option execute_on = final

### DISCUSSION THREAD

**Discussion by doquang** [idaholab/moose#19947] [Q&A General] [answered]
Dear Moosers,

I have an unusual situation, can someone explain. Thanks in advance,

--- 
Run the original test `moose/test/tests/vectorpostprocessors/elements_along_line/2d.i`, I got a correct data on csv files.
Run the modified above test as:
```
[VectorPostprocessors]
  [./elems]
    type = ElementsAlongLine
    start = '0.05 0.05 0'
    end = '0.05 0.405 0'
    **execute_on = final**
  [../]
[]
```
I got an empty csv file.

**Comment by GiudGiud** [comment, marked as answer]
Hello

What's happening is that the postprocessor is correctly executed on FINAL but the outputting is not.
Just change the [Outputs] to 
```
[Outputs]
  exodus = true
  csv = true
  execute_on = 'final'
[]
```

Guillaume

**Reply by doquang** [reply]
Thanks @GiudGiud . It works!
